internal/feedback/feedback.go

```diff
@@ -82,6 +82,55 @@ func (cf ConstraintFeedback) LogFeedback(logger *log.Logger) {
 	}
 }
 
+type brokenImport struct {
+	projectPath string
+	version     *gps.StringDiff
+	revision    *gps.StringDiff
+}
+
+func (bi brokenImport) String() string {
+	var pr string
+	var cr string
+	if bi.revision != nil {
+		pr = fmt.Sprintf("(%s)", trimSHA(bi.revision.Previous))
+		cr = fmt.Sprintf("(%s)", trimSHA(bi.revision.Current))
+	}
+
+	return fmt.Sprintf("%v %s for %s. Locking in %v %s",
```

Comment:
With the inclusion of branch and source, maybe something like this would do:

```
Warning: Unable to preserve imported lock VERSION/BRANCH (REV) for PROJECT. Locking in VERSION/BRANCH (REV)
```

Where for `VERSION/BRANCH` we display whichever is not an empty string.

If the source diff is not nil:

```
Warning: Unable to preserve imported lock VERSION/BRANCH (REV) for PROJECT(SOURCE). Locking in VERSION/BRANCH (REV) for PROJECT(SOURCE)
```

Basically, don't print the source unless it's changed.